icubam/www/handlers/home.py

```diff
@@ -22,45 +24,71 @@ class HomeHandler(base.BaseHandler):
 
   ROUTE = '/'
   POPUP_TEMPLATE = 'popup.html'
+  # TODO(olivier): put this in the config
   CLUSTER_KEY = 'dept'  # city
 
   def initialize(self, config, db):
     super().initialize(config, db)
     loader = tornado.template.Loader(self.get_template_path())
     self.token_encoder = token.TokenEncoder(self.config)
     self.popup_template = loader.load(self.POPUP_TEMPLATE)
-    self.icus_df = self.db.get_icus()
-
-  def get_city_data(self):
-    coords = dict()
-    cluster_id = dict()  # the city for each icu
-    for city, rows in self.icus_df.groupby(self.CLUSTER_KEY):
-      coords[city] = {'lat': rows.lat.mean(), 'lng': rows.long.mean()}
-      for icuid in rows.icu_id.to_list():
-        cluster_id[icuid] = city
-    return coords, cluster_id
-
-  def get_phones(self):
+
+  def groupby(self):
     result = collections.defaultdict(list)
-    for index, row in self.icus_df.iterrows():
-      result[row['icu_id']].append(row['telephone'])
+    for icu in self.icus.values():
```

Comment:
With the function that coverts DB objects to dicts, I'm curious if this may also work for grouping?

https://stackoverflow.com/questions/20638006/convert-list-of-dictionaries-to-a-pandas-dataframe

Reply:
that is a good point. the function is not recursive for now. let's keep it for later though, since there are already a lot in the PR